packages/expo-cli/src/__tests__/exp-test.ts

```diff
@@ -0,0 +1,82 @@
+import program, { Command } from 'commander';
+import { UnifiedAnalytics } from 'xdl';
+
+import { registerCommands } from '../../build/commands';
+import { trackUsage } from '../../build/exp';
+
+Command.prototype.helpGroup = jest.fn().mockImplementation(_ => program);
+Command.prototype.asyncActionProjectDir = jest.fn().mockImplementation(_ => program);
+Command.prototype.longDescription = jest.fn().mockImplementation(_ => program);
+Command.prototype.asyncAction = jest.fn().mockImplementation(_ => program);
+Command.prototype.allowOffline = jest.fn().mockImplementation(_ => program);
+Command.prototype.urlOpts = jest.fn().mockImplementation(_ => program);
+
+describe(trackUsage.name, () => {
+  const commands = generateCommandJSON();
+  const commandNames = commands.map(cmd => cmd.name);
+  const commandAliases = commands
+    .filter((cmd: CommandMockData) => cmd._alias)
+    .map(cmd => cmd._alias);
+
+  const spy = jest.spyOn(UnifiedAnalytics, 'logEvent');
+
+  beforeEach(() => {
+    spy.mockClear();
+  });
+
+  it.each(commandNames)('matches on command name: %s', name => {
+    process.argv[2] = name;
+
+    trackUsage(commands as any);
+
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it.each(commandAliases)('matches on command alias: %s', alias => {
+    process.argv[2] = alias;
+
+    trackUsage(commands as any);
+
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('does not match on undefined aliases', () => {
+    process.argv[2] = undefined;
+
+    trackUsage(commands as any);
+
+    expect(spy).not.toHaveBeenCalled();
+  });
+
+  it('does not match on empty string', () => {
+    process.argv[2] = '';
+
+    trackUsage(commands as any);
+
+    expect(spy).not.toHaveBeenCalled();
+  });
+});
+
+type CommandMockData = {
+  name: string;
+  _name: string;
+  description: string;
+  alias?: string;
+  _alias?: string;
+};
+
+function generateCommandJSON(): CommandMockData[] {
```

Comment:
copied from introspect but slightly adjusted to contain `_name` and `_alias` and not contain `options` or `group`. figured the use-case was different enough to warrant a new function rather than adjusting the one present in introspect.